docs/data/toolpad/core/components/dashboard-layout/DashboardLayoutCustomPageItems.tsx

```diff
@@ -0,0 +1,214 @@
+  },
+  {
+    segment: 'dashboard',
+    title: 'Dashboard',
+    icon: <DashboardIcon />,
+  },
+  {
+    segment: 'external',
+    title: 'External Link',
+    icon: <LinkIcon />,
+  },
+  {
+    segment: 'selected',
+    title: 'Selected Item',
+    icon: <CheckBoxIcon />,
+  },
+  {
+    segment: 'disabled',
+    title: 'Disabled Item',
+    icon: <DisabledByDefaultIcon />,
+  },
+  {
+    segment: 'hidden',
+    title: 'Hidden Item',
+    icon: <VisibilityOffIcon />,
+  },
+  {
+    segment: 'expanded',
+    title: 'Expanded Folder',
+    icon: <FolderIcon />,
+    children: [
+      {
+        segment: 'sub-item-1',
+        title: 'Sub-Item 1',
+        icon: <DescriptionIcon />,
+      },
+      {
+        segment: 'sub-item-2',
+        title: 'Sub-Item 2',
+        icon: <DescriptionIcon />,
+      },
+    ],
+  },
+  {
+    segment: 'custom',
+    title: 'Custom Item',
+  },
+];
+
+function CustomPageItem({
+  item,
+  mini,
+}: {
+  item: NavigationPageItem;
+  mini: boolean;
+}) {
+  return (
+    <ListItem
+      sx={(theme) => ({
+        color: theme.palette.secondary.main,
+        overflowX: 'hidden',
+      })}
+    >
+      {mini ? (
+        <IconButton
+          aria-label="custom"
+          sx={(theme) => ({
+            color: theme.palette.secondary.main,
+          })}
+        >
+          <AutoAwesomeIcon />
+        </IconButton>
+      ) : (
+        <ListItemButton>
+          <ListItemIcon
+            sx={(theme) => ({
+              color: theme.palette.secondary.main,
+            })}
+          >
+            <AutoAwesomeIcon />
+          </ListItemIcon>
+          <ListItemText
+            primary={item.title}
+            sx={{
+              whiteSpace: 'nowrap',
+            }}
+          />
+        </ListItemButton>
+      )}
+    </ListItem>
+  );
+}
+
+const demoTheme = createTheme({
+  cssVariables: {
+    colorSchemeSelector: 'data-toolpad-color-scheme',
+  },
+  colorSchemes: { light: true, dark: true },
+  breakpoints: {
+    values: {
+      xs: 0,
+      sm: 600,
+      md: 600,
+      lg: 1200,
+      xl: 1536,
+    },
+  },
+});
+
+function DemoPageContent({ pathname }: { pathname: string }) {
+  return (
+    <Box
+      sx={{
+        py: 4,
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        textAlign: 'center',
+      }}
+    >
+      <Typography>Dashboard content for {pathname}</Typography>
+    </Box>
+  );
+}
+
+interface DemoProps {
+  /**
+   * Injected by the documentation to work in an iframe.
+   * Remove this when copying and pasting into your project.
+   */
+  window?: () => Window;
+}
+
+export default function DashboardLayoutCustomPageItems(props: DemoProps) {
+  const { window } = props;
+
+  const router = useDemoRouter('/dashboard');
+
+  // Remove this const when copying and pasting into your project.
+  const demoWindow = window !== undefined ? window() : undefined;
+
+  const renderPageItem = React.useCallback(
```

Comment:
I think it makes sense to add this renderPageItem definition to the demo preview as well

```suggestion
// preview-start
  const renderPageItem = React.useCallback(
```